gems/workflow/app/services/workflow/instance/step/complete.rb

```diff
@@ -12,9 +12,20 @@ def initialize(step)
       end
 
       def run
+        process = @step.process
+        prev_process_status = process.status
+
         @step.completed = true
         @step.completed_at = DateTime.now
         @step.save!
+
+        if process.status == Workflow::Instance::Process::DONE
+          process.completed_at = DateTime.now
+        end
```

Comment:
sorry not following here.  don't you have to check if the step you completed was the last step of the process? if so then mark process completed?

Reply:
the `Process#status` function will do that

Reply:
got it! thx

Reply:
I think status is a useful front end concept that we should put in the serializer.  but in the back end I want to explicitly call out two separate things: your completeness and whether your dependencies are met are not.

I think of serializers as "presentation logic" where you talk in more end user terms.  and I try to keep the rest of the code more engineering-level precise.